common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationActivityFactory.java

```diff
@@ -77,7 +77,7 @@ public static Intent getAuthorizationActivityIntent(final Context context,
         final LibraryConfiguration libraryConfig = LibraryConfiguration.getInstance();
         if (ProcessUtil.isBrokerProcess(context)) {
             intent = new Intent(context, BrokerAuthorizationActivity.class);
-        } else if (libraryConfig.isAuthorizationInCurrentTask()) {
+        } else if (libraryConfig.isAuthorizationInCurrentTask() && !authorizationAgent.equals(AuthorizationAgent.WEBVIEW)) {
```

Comment:
A comment here in the code explaining why this is the case would be super-helpful.  It'll be almost impossible to reverse-engineer this from the code.

Reply:
+1

It is not clear how this change solves that crash. It probably does avoid the crash altogether by not even taking that code path, but I'm thinking if we can still take this code path in a safe way and not crash. Do we know what actually caused the crash?

Reply:
Yes.  The crash happens when the code attempts to use the default browser, but none is available on the device.  In that situation we end up providing the incorrect fragment to the authorization activity... which throws illegal state exception.  This code ensures that if we have to fallback to webview... because no browser is installed.... then we'll use the correct authorization activity and fragment.